src/core.js

```diff
@@ -392,6 +393,99 @@ define([
         ];
     };
 
+    fn._init_bulk_update_questions = function () {
+        var _this = this,
+            menuItems = this.getBulkUpdateMenuItems();
+
+        var $lastItem = this.$f.find('.fd-bulk-update-header');
+        $lastItem.nextUntil(".divider").remove();
+        _(menuItems).each(function (menuItem) {
+            var $menuLink = $(util.format(
+                    "<a tabindex='-1' href='#'>{name}</a>",
+                    _.extend(menuItem)
+                )).click(function (e) {
+                    e.preventDefault();
+                    _this.showConfirmBulkActionModal(menuItem.confirmMessage, menuItem.confirmAction);
+                }),
+                $newItem = $("<li></li>").append($menuLink);
+            $lastItem.after($newItem);
+            $lastItem = $newItem;
+        });
+
+    };
+
+    fn.getBulkUpdateMenuItems = function () {
+        var _this = this,
+            menuItems =  [
+                {
+                    name: gettext('Make Required'),
+                    confirmMessage: gettext("You are about to make all existing " +
+                        "questions on this form required. This action will overwrite " +
+                        "the current configuration of this form and can only be " +
+                        "undone by editing individual questions."),
+                    confirmAction: function () {
+                        _this.makeAllQuestionsRequired();
+                    },
+                },
+            ];
+
+        if (_this.data.core.databrowser) {
+            menuItems.push({
+                name: gettext('Load Default Values'),
+                confirmMessage: gettext("You are about to set matching case " +
+                    "properties as Default Values for all existing questions. " +
+                    "This action will overwrite existing Default Values for " +
+                    "all questions with matching case properties and can only " +
+                    "be undone by editing individual questions."),
+                confirmAction: function () {
+                    _this.defaultMatchingQuestionsToCaseProperties();
+                },
+            });
+        }
+        return menuItems;
+    };
+
+    fn.makeAllQuestionsRequired = function () {
+        var _this = this;
+        _this.data.core.form.walkMugs(function (mug) {
+            if (mug.spec.requiredAttr && mug.isVisible('requiredAttr')) {
+                mug.p.requiredAttr = true;
+            }
+        });
+        _this.refreshCurrentMug();
+    };
+
+    fn.defaultMatchingQuestionsToCaseProperties = function () {
+        var _this = this,
+            caseProperties = _.keys(_this.datasources.getHashtagMap());
+        _this.data.core.form.walkMugs(function (mug) {
+            if (mug.isVisible('defaultValue')) {
+                var caseProp = '#case/' + mug.p.nodeID;
+                if (caseProperties.indexOf(caseProp) !== -1) {
```

Comment:
`indexOf()` is O(n), while [`hasOwnProperty()` is many times faster if not O(1)](https://stackoverflow.com/a/55057332/10840)
```suggestion
            caseProperties = _this.datasources.getHashtagMap({});
        _this.data.core.form.walkMugs(function (mug) {
            if (mug.isVisible('defaultValue')) {
                var caseProp = '#case/' + mug.p.nodeID;
                if (caseProperties.hasOwnProperty(caseProp)) {
```